app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -71,28 +128,164 @@ class PopupView : FrameLayout {
+                            Logger.e(
+                                "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                                    ee
+                                )
+                            )
+                        }
+                    }
+                }
+            } catch (ie: InterruptedException) {
+                Logger.e(
+                    "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                        ie
+                    )
                 )
-            )
+            } catch (ee: ExecutionException) {
+                Logger.e(
+                    "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                        ee
+                    )
+                )
+            }
+        }
+    }
+
+    /**
+     * Updates the given popup field value with the appropriately cast string value.
+     *
+     * @param manager - PopupManager
+     * @param field   - PopupField
+     * @param value   - String
+     * @return - ArcGISRuntimeException if there's a problem updating the field value,
+     * otherwise null.
+     */
+    private fun updateValue(
+        manager: PopupManager,
+        field: PopupField?,
+        value: String?
+    ): ArcGISRuntimeException? {
+        var error: ArcGISRuntimeException? = null
+
+        when (manager.getFieldType(field)) {
+            Field.Type.SHORT -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toShort(), field)
+            }
+            Field.Type.INTEGER -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toInt(), field)
+            }
+            Field.Type.FLOAT -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toFloat(), field)
+            }
+            Field.Type.DOUBLE -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toDouble(), field)
+            }
+            Field.Type.DATE -> {
+                val calendar = Calendar.getInstance()
+                // Save any non-empty dates
+                if (value!!.trim { it <= ' ' }.isNotEmpty()) {
+                    try {
+                        val date = SimpleDateFormat(
+                            "M/dd/yy hh:m a",
+                            Locale.getDefault()
+                        ).parse(value)
+                        calendar.timeInMillis = date.time
```

Comment:
The compiler gives a warning here `Unsafe use of a nullable receiver of type Date?`
But I think this can be null, so we can just do: 
```suggestion
                        calendar.timeInMillis = date!!.time
```

Reply:
I have removed this code. No longer applicable.